beacon-chain/light-client/store.go

```diff
@@ -137,13 +145,13 @@ func (s *Store) validateUpdate(update *Update,
 	// saved in the state of attested header. Note that the genesis finalized checkpoint root is represented as a zero
 	// hash.
 	if !update.isFinalityUpdate() {
-		if update.GetFinalizedHeader() != &(ethpbv1.BeaconBlockHeader{}) {
+		if update.GetFinalizedHeader() != nil {
 			return errors.New("finality branch is present but update is not finality")
 		}
 	} else {
 		var finalizedRoot [32]byte
 		if update.GetFinalizedHeader().Slot == genesisSlot {
-			if update.GetFinalizedHeader() != &(ethpbv1.BeaconBlockHeader{}) {
+			if update.GetFinalizedHeader() != nil {
```

Comment:
This is the branch for `finalityHeader == genesis` case, so it won't be `nil`.

```python
        if update.finalized_header.slot == GENESIS_SLOT:
            assert update.finalized_header == BeaconBlockHeader()
            finalized_root = Bytes32()
```
I think we can compare
     `*update.GetFinalizedHeader() == ethpbv1.BeaconBlockHeader{}`  (compare all 0s in memory)

Reply:
that yields

```
./store.go:154:8: invalid operation: *update.GetFinalizedHeader() != ethpbv1.BeaconBlockHeader{} (struct containing impl.MessageState cannot be compared)
```

perhaps `String()`?

Reply:
Let's do it.  (Sounds good)